### PR TITLE
Ensure wigner_3j compatibility with numpy floats

### LINKED ISSUE #27219: wigner_3j Violating numpy behavioral subtyping
Dear Developers,

I am currently using the most up to date versions of numpy (2.1.3) and sympy (1.13.3). I get an error when I run the following code snippet.

```
import numpy as np
from sympy.physics.wigner import wigner_3j

y = np.float64(6.0)
x = wigner_3j(2, y, 4.0, 0, 0, 0)
```

Error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[5], line 5
      2 from sympy.physics.wigner import wigner_3j
      4 y = np.float64(6.0)
----> 5 x = wigner_3j(2, y, 4.0, 0, 0, 0)

File ~\code\.venv\jlab\lib\site-packages\sympy\physics\wigner.py:219, in wigner_3j(j_1, j_2, j_3, m_1, m_2, m_3)
    130 def wigner_3j(j_1, j_2, j_3, m_1, m_2, m_3):
    131     r"""
    132     Calculate the Wigner 3j symbol `\operatorname{Wigner3j}(j_1,j_2,j_3,m_1,m_2,m_3)`.
    133 
   (...)
    216     - Jens Rasch (2009-03-24): initial version
    217     """
--> 219     j_1, j_2, j_3, m_1, m_2, m_3 = map(_int_or_halfint,
    220                                        [j_1, j_2, j_3, m_1, m_2, m_3])
    222     if m_1 + m_2 + m_3 != 0:
    223         return S.Zero

File ~\code\.venv\jlab\lib\site-packages\sympy\physics\wigner.py:127, in _int_or_halfint(value)
    125 elif isinstance(value, Float):
    126     return _int_or_halfint(float(value))
--> 127 raise ValueError("expecting integer or half-integer, got %s" % value)

ValueError: expecting integer or half-integer, got 6.0
```

It appears that wigner_3j violates behavioral subtyping/no longer accepts numpy floats or ints. This issue appears to be similar to issue #27012, and was fixed in pr #27066. Can this also be fixed for the wigner_3j function too?

Thanks,
Simon

### PR DESCRIPTION
**Pull request by RushabhMehta2005**
#### References to other Issues or PRs

Fixes #27219 

#### Brief description of what is fixed or changed
Modified the wigner_3j function to ensure compatibility with numpy floats by adding calls to sympify for the input params. This solves the issue where it previously raised a ValueError when encountered with a numpy float.

#### Release Notes

* physics.wigner
  * Numpy floats can be used with wigner_3j

### CODE REVIEW

File: sympy/physics/wigner.py [hunk 1]
```diff
@@ -216,8 +216,12 @@ def wigner_3j(j_1, j_2, j_3, m_1, m_2, m_3):
     - Jens Rasch (2009-03-24): initial version
     """
 
-    j_1, j_2, j_3, m_1, m_2, m_3 = map(_int_or_halfint,
-                                       [j_1, j_2, j_3, m_1, m_2, m_3])
+    j_1 = sympify(j_1)
```
Comment (oscarbenjamin) [inline sympy/physics/wigner.py:219]: Does this still check whether the value is an integer or a half-integer?

**Comment by oscarbenjamin** [comment]
A test should be added for what is fixed. It can be part of the same test function that was added in https://github.com/sympy/sympy/pull/27066/files

**Comment by oscarbenjamin** [comment]
The test can be added in a new commit and pushed to this PR (no need to close and open a new PR).

File: sympy/physics/wigner.py (hunk 1 above)
Comment (RushabhMehta2005) [reply sympy/physics/wigner.py:219]: On manual testing it works for both integers and half integers.

**Comment by RushabhMehta2005** [comment]
> A test should be added for what is fixed. It can be part of the same test function that was added in https://github.com/sympy/sympy/pull/27066/files

The new function added in this PR is named "test_clebsch_gordan_numpy", should I add my new test here by using the relation between wigner_3j and the Clebsch Gordan coefficient? I can add it there or in the test_wigner function in the same file.

File: sympy/physics/wigner.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/physics/wigner.py:219]: What if the value is not an integer or half integer? The purpose of the `_int_or...` function is to reject values that are not half integers.

**Comment by oscarbenjamin** [comment]
> named "test_clebsch_gordan_numpy

You can rename the function if you want but just add the test there. We don't need a separate function for this.

File: sympy/physics/wigner.py (hunk 1 above)
Comment (RushabhMehta2005) [reply sympy/physics/wigner.py:219]: Oh, the expected behaviour in the documentation is for it to throw a ValueError, right now my code is returning 0 for these invalid cases. I need to further improve on this

**Comment by RushabhMehta2005** [comment]
Okay, so I will add 1 or 2 tests for the wigner_3j function there.

File: sympy/physics/wigner.py (hunk 1 above)
Comment (RushabhMehta2005) [reply sympy/physics/wigner.py:219]: New commits have been added.

**Comment by RushabhMehta2005** [comment]
Please review this! Sorry for the failed check due to trailing whitespace.

File: sympy/physics/wigner.py (hunk 1 above)
Comment (RushabhMehta2005) [reply sympy/physics/wigner.py:219]: All three cases - integers, half integers and neither ints or half ints are now handled, the function raises a ValueError as expected from the docs in the lattermost case. 3 tests are added to cover the error raising, the numpy float and half ints as well.

**Comment by RushabhMehta2005** [comment]
Hello @oscarbenjamin, my PR has passed all checks and is ready for review whenever it is convienient for you. Thank you for your time and patience with me.

File: sympy/physics/wigner.py [hunk 2]
```diff
@@ -216,8 +216,14 @@ def wigner_3j(j_1, j_2, j_3, m_1, m_2, m_3):
...
+    j_2 = sympify(j_2)
+    j_3 = sympify(j_3)
+    m_1 = sympify(m_1)
+    m_2 = sympify(m_2)
+    m_3 = sympify(m_3)
+
+    j_1, j_2, j_3, m_1, m_2, m_3 = map(_int_or_halfint, [j_1, j_2, j_3, m_1, m_2, m_3])
```
Comment (smichr) [inline sympy/physics/wigner.py:226]: You can delete all the previous lines that were added and write
```python
j_1, j_2, j_3, m_1, m_2, m_3 = \
    map(_int_or_halfint, map(sympify,
        [j_1, j_2, j_3, m_1, m_2, m_3]))
```

Comment (RushabhMehta2005) [reply sympy/physics/wigner.py:226]: Alright, I'll do it by EOD! Thank you for the review.

**Pull request merged**